components/script/script_module.rs

```diff
@@ -1214,10 +1216,26 @@ pub unsafe fn EnsureModuleHooksInitialized(rt: *mut JSRuntime) {
 
     SetModuleResolveHook(rt, Some(HostResolveImportedModule));
     SetModuleMetadataHook(rt, Some(HostPopulateImportMeta));
-    SetScriptPrivateReferenceHooks(rt, None, None);
+    SetScriptPrivateReferenceHooks(
+        rt,
+        Some(host_add_ref_top_level_script),
+        Some(host_release_top_level_script),
+    );
     SetModuleDynamicImportHook(rt, Some(host_import_module_dynamically));
 }
 
+#[allow(unsafe_code)]
+unsafe extern "C" fn host_add_ref_top_level_script(value: *const Value) {
+    let val = Rc::from_raw(value);
```

Comment:
We need to use `value.to_private() as *const ModuleScript` here.

Reply:
ok! 🙆‍♂️ Thanks for the tip!

But, I'm don't clearly understand why we need to do the `to_private` and `as *const ModuleScript` here? Does that mean using `to_private` and cast to a raw pointer of `ModuleScript` is correctly pointed to the same one we saved before?

Reply:
Correct.

Reply:
I see! Thanks! Always happy to learn more :D

Reply:
hmm, wait, this commit doesn't compile! Let me fix it!